### PR TITLE
Code Review: Bare Metal Forms and Helpers

### PR DESCRIPTION
**Pull request by umairarshadbutt**
This project includes multiple ways to build a form in rails that can send parameters to a controller so that it can build an instance of a model with those parameters.
In the project, we practiced the following concepts of rails
- Model
- Migration
- Controller
- View
- form_tag
- form_for
- errors

### CODE REVIEW

**Review by davidmrtz-dev — CHANGES_REQUESTED** [review]
## Hi @umairarshadbutt  @epinczinger  👨‍🚀 
### Very good job so far.

However, there are some issues that need to be addressed before moving on.  👨‍💻  ⬇️ 
Please check out below.

****
Users new view.
- [x] uses users_path helper for action
On the first form, you are using action="/users", please change it in order to use "users_path" helper.
- [x] Uses form_authenticity_token
We already have the input for the password, but we also have to send the authenticity token.
Consider the following example: 
```ruby
<input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>">
```


(screenshot screenshot from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
****
- [x] Form using "form_tag"
There's is no form comment using the pointed form, kindly add the required form.

(screenshot from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
****
### **Please make the changes required and submit for re-review once you are ready!** 

****

### Status: Changes required. ♻️  
****

Happy coding!  💻  
Have a great day 🙂

If you need more clarifications kindly reach me on slack.
**[TSE:  David Elí]** 
Slack: @ David Eli Martinez Garcia

**Comment by umairarshadbutt** [comment]
The review was very helpful to understanging and revising the concepts

File: app/views/users/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    <br />
+    <label for="email">Password:</label>
+    <input type="password" id="password" name="user[password]" /><br />
+    <br />
+    <input type="submit" value="Submit" />
+</form>
+
+<%= form_tag("/users_path", method: "post") do %>
+    <%= label_tag(:username, "Username:") %>
+    <%= text_field_tag(:username) %><br>
```
Comment (kobiyoyo) [inline app/views/users/edit.html.erb]: This should be written like this 
```html
  <%= label_tag 'username' %>
  <%= text_field_tag 'username' %>
````
Note: it should be implemented for all fields .

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    <label for="email">Email:</label>
+    <input type="email" id="email" name="user[email]" /><br />
+    <br />
+    <label for="email">Password:</label>
+    <input type="password" id="password" name="user[password]" /><br />
+    <br />
+    <input type="submit" value="Submit" />
+</form>
+
+<%= form_tag("/users_path", method: "post") do %>
```
Comment (kobiyoyo) [inline app/views/users/edit.html.erb]: You can write it this way, you don't need a slash before `users_path`
Also this isnt required `method: post`
`<%= form_tag(users_path) do %>`

File: app/views/users/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<!--
+<form_tag accept-charset="UTF-8" action="/users_path" method="post">
```
Comment (kobiyoyo) [inline app/views/users/edit.html.erb]: Remove the slash before `users_path`

File: app/views/users/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    <%= label_tag(:email, "email:") %>
+    <%= text_field_tag(:email) %><br>
+
+    <%= label_tag(:password, "password:") %>
+    <%= password_field_tag(:password) %><br>
+
+    <%= submit_tag("Submit") %>
+<% end %>
+-->
```
Comment (kobiyoyo) [inline app/views/users/edit.html.erb]: This should be commented out properly

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [x] Check comments below  

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: app/views/users/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            <h2><%= pluralize(@user.errors.count, "error") %> 
+            prohibited this user from being edited</h2>
+
+            <ul>
+                <% @user.errors.full_messages.each do |msg| %>
+                    <li><%= msg %></li>
+                <% end %>
+            </ul>
+        </div>
+    <% end %>
```
Comment (kobiyoyo) [inline app/views/users/edit.html.erb:46]: Let's remove this as it is not requirement, but you can implement this in future projects

File: app/views/users/edit.html.erb (hunk 1 above)
Comment (epinczinger) [reply app/views/users/edit.html.erb]: Done!

File: app/views/users/edit.html.erb (hunk 2 above)
Comment (epinczinger) [reply app/views/users/edit.html.erb]: Done!

File: app/views/users/edit.html.erb (hunk 3 above)
Comment (epinczinger) [reply app/views/users/edit.html.erb]: Done!

File: app/views/users/edit.html.erb (hunk 4 above)
Comment (epinczinger) [reply app/views/users/edit.html.erb]: Done!

File: app/views/users/edit.html.erb (hunk 5 above)
Comment (epinczinger) [reply app/views/users/edit.html.erb:46]: Done!

**Comment by epinczinger** [comment]
All changes have been done.
Thanks for the review! It was very helpful

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Thank you for applying the suggested changes.
You can merge it now :+1: 

Good continuation :handshake: 

Mael FOSSO

**Pull request merged**